core/shared/src/main/scala/fs2/interop/flow/StreamSubscriber.scala

```diff
@@ -268,10 +271,6 @@ private[flow] final class StreamSubscriber[F[_], A] private (
         case Idle(s) =>
           WaitingOnUpstream(cb, s) -> run {
             s.request(chunkSize.toLong)
-            // We do the updates here,
-            // to ensure they happen after we have secured the state.
-            inOnNextLoop = false
-            index = 0
```

Comment:
Redundant.
The initial state is this, and finishing each loop leaves the state like this as well.